l2geth/eth/api_backend.go

```diff
@@ -307,10 +307,6 @@ func (b *EthAPIBackend) SendTx(ctx context.Context, signedTx *types.Transaction)
 			if *to == (common.Address{}) {
 				return errors.New("Cannot send transaction to zero address")
 			}
-			// Prevent transactions from being submitted if the gas limit too high
-			if signedTx.Gas() >= b.gasLimit {
-				return fmt.Errorf("Transaction gasLimit (%d) is greater than max gasLimit (%d)", signedTx.Gas(), b.gasLimit)
-			}
```

Comment:
Could be a good idea that we don't remove this check entirely, but instead just replace it with a call to `DoEstimateGas` + a ~50% safety margin. That way we don't end up with a tx that is way incorrectly priced

Reply:
I'm open if you don't think this is worth the complexity

Reply:
+1 on this idea

Reply:
If we add this before the execution price goes into the state and the sequencer changes the values then this will brick a lot things and will require manual gasLimits to be set

Reply:
There is currently a bug in the gas estimation logic that is unknown. The only way that this would be safe is ignore the case when there is an error, otherwise a set of transactions will not be able to be sent

Reply:
I think we should follow up with this addition after we figure out the bug in gas estimation

Reply:
I've opened up an issue here: https://github.com/ethereum-optimism/optimism/issues/694

Reply:
Ok I'm down to do this! Let's make sure to prioritize the follow up though